hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -127,6 +127,10 @@ public static String offsetsToStr(OffsetRange[] ranges) {
               exhaustedPartitions.add(range.partition());
             }
             allocedEvents += toOffset - range.untilOffset();
+            // We need recompute toOffset if allocaedEvents larger than numEvents.
+            if (allocedEvents > numEvents) {
+              toOffset = (range.untilOffset() + eventsPerPartition) + (numEvents - allocedEvents);
+            }
```

Comment:
I think we still need the Math.min(..) check here to handle the case where toOffset is being set to the maximum range (partitions can have skew for eg.) in L125 above.. 

I am also wondering if introducing a new variable like below will fix the issue (top of my head)
```
long offsetsToAdd = Math.max(eventsPerPartition, (numEvents - allocedEvents))
long toOffset. = Math.min(toOffsetMax,  range.untilOffset() + offsetsToAdd)
```

and keep the flow simpler.